src/Analyser/TypeSpecifier.php

```diff
@@ -2140,6 +2097,70 @@ public function resolveIdentical(Expr\BinaryOp\Identical $expr, Scope $scope, Ty
 		}
 		$rightType = $scope->getType($rightExpr);
 
+		if (
+			!$context->null()
+			&& $unwrappedLeftExpr instanceof FuncCall
+			&& count($unwrappedLeftExpr->getArgs()) >= 1
+			&& $unwrappedLeftExpr->name instanceof Name
+			&& in_array(strtolower((string) $unwrappedLeftExpr->name), ['count', 'sizeof'], true)
+			&& $rightType->isInteger()->yes()
```

Comment:
use a proper `Type` method instead of `&& $constantType instanceof ConstantIntegerType`